Rename 'foo-procedures' to 'banana-procedures'

Mythicaeda opened the pull request:
* **Tickets addressed:** None
* **Review:** By commit  <!-- Choose from: "by commit", "by file" -->
* **Merge strategy:** Merge (no squash)  <!-- Choose from: "merge (no squash)", "squash and merge" -->

## Description
<!-- What approach was taken to satisfy the ticket being addressed? What should reviewers be aware of? -->
These procedures use the Bananation mission model, not the Foo mission model. The old module name was confusing.

## Verification
<!-- How were the changes validated? Were any automated tests added, updated, removed, or re-baselined? -->
I was able to build the module using the command in the README. I also did a project-wide search and found no other instances of either `foo-procedures` or `fooprocedures`

## Documentation
<!-- What documentation was invalidated by these changes? Which artifacts should reviewers check for accuracy and completeness? -->
I saw no mention of this module in the Aerie Docs website and have updated the README to use the new name.

Mythicaeda committed:
Rename 'foo-procedures' to 'banana-procedures'

These procedures use the Bananation mission model, not the Foo mission model

JoelCourtney commented on procedural/examples/banana-procedures/src/main/java/gov/nasa/ammos/aerie/procedural/examples/bananaprocedures/constraints/Hello.java:
I think this is pretty much useless at this point. We used it as an early test of jar building before we had actual goal classes. Can you delete it please

Mythicaeda replied:
Done!

JoelCourtney reviewed (approved):
I have a request, but I'm approving anyway because its not actually on the merits of the PR

mattdailis reviewed (approved):
Good call

Mythicaeda committed:
Remove Hello.java

Was used as an early test of Jar building before there were goal classes, but is not needed anymore.
